casper/src/main/scala/coop/rchain/casper/genesis/Genesis.scala

```diff
@@ -103,6 +94,62 @@ object Genesis {
                      )
     } yield genesisBlock
 
+  private def getVaults[F[_]: Sync: Log: RaiseIOError](
+      maybeVaultPath: Option[String],
+      defaultVaultPath: Path
+  ): F[Seq[Vault]] =
+    maybeVaultPath match {
+      case Some(vaultsPathStr) =>
+        val vaultsPath = Paths.get(vaultsPathStr)
+        Monad[F].ifM(exists(vaultsPath))(
+          vaultFromFile[F](vaultsPath),
+          Sync[F].raiseError(new Exception(s"Specified vaults file $vaultsPath does not exist"))
+        )
+      case None =>
+        Monad[F].ifM(exists(defaultVaultPath))(
+          Log[F].info(s"Using default file $defaultVaultPath") >> vaultFromFile[F](
+            defaultVaultPath
+          ),
+          Log[F]
+            .warn(
+              "No vaults file specified and no default file found. No vaults will exist at genesis."
+            )
+            .map(_ => Seq.empty[Vault])
+        )
+    }
+
+  private def vaultFromFile[F[_]: Sync: Log: RaiseIOError](vaultsPath: Path): F[Seq[Vault]] =
+    for {
+      lines <- SourceIO
+                .open(vaultsPath)
+                .use(_.getLines)
+                .adaptError {
+                  case ex: Throwable =>
+                    new RuntimeException(
+                      s"Failed to read ${vaultsPath.toAbsolutePath} for reason: ${ex.getMessage}"
+                    )
+                }
+      vaults <- lines.traverse(parseLine[F])
+    } yield vaults
+
+  private def parseLine[F[_]: Sync: Log: RaiseIOError](line: String): F[Vault] =
+    Sync[F].fromEither(
+      fromLine(line)
+        .leftMap(errMsg => new RuntimeException(s"Error in parsing vaults file: $errMsg"))
+    )
+
+  private def fromLine(line: String): Either[String, Vault] = line.split(",") match {
+    case Array(ethAddress, initRevBalanceStr, _) =>
+      Try(BigInt(initRevBalanceStr)) match {
+        case Success(initRevBalance) =>
+          Right(Vault(RevAddress.fromEthAddress(ethAddress), initRevBalance.toLong)) // TODO: Remove BigInt conversion
```

Comment:
I don't understand the motivation behind introducing the `BigInt` two lines above and adding a todo note to remove it here. Since you need a `long` at the very end why not do `initRevBalanceStr.toLong`?